From 8711f11411915a04e9fe8e4d6226206d00be3a59 Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arunsr@codeaurora.org>
Date: Tue, 29 May 2018 09:03:33 +0530
Subject: [PATCH 1/3] audioringbuffer: Don't calculate bytes per sample unless
 needed

This avoids accessing the channel count in the ringbuffer spec if it is
not needed. Specifically, this is needed for compressed formats, where
reordering is not possible and a channel count might not be available.
---
 gst-libs/gst/audio/gstaudioringbuffer.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/audio/gstaudioringbuffer.c b/gst-libs/gst/audio/gstaudioringbuffer.c
index 13f4d088e..d3c13152f 100644
--- a/gst-libs/gst/audio/gstaudioringbuffer.c
+++ b/gst-libs/gst/audio/gstaudioringbuffer.c
@@ -1783,9 +1783,11 @@ gst_audio_ring_buffer_read (GstAudioRingBuffer * buf, guint64 sample,
   segtotal = buf->spec.segtotal;
   channels = buf->spec.info.channels;
   bpf = buf->spec.info.bpf;
-  bps = bpf / channels;
   sps = buf->samples_per_seg;
 
+  if (need_reorder)
+    bps = bpf / channels;
+
   to_read = len;
   /* read enough samples */
   while (to_read > 0) {
-- 
2.18.0

